pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -1452,216 +1452,93 @@ protected void internalGetPartitionedStatsInternal(AsyncResponse asyncResponse,
         });
     }
 
-    protected void internalDeleteSubscription(AsyncResponse asyncResponse,
-                                              String subName, boolean authoritative, boolean force) {
-        if (force) {
-            internalDeleteSubscriptionForcefully(asyncResponse, subName, authoritative);
-        } else {
-            internalDeleteSubscription(asyncResponse, subName, authoritative);
-        }
-    }
-
-    protected void internalDeleteSubscription(AsyncResponse asyncResponse, String subName, boolean authoritative) {
+    protected void internalDeleteSubscriptionAsync(AsyncResponse asyncResponse, String subName, boolean authoritative,
```

Comment:
It's better to avoid passing the AsyncResponse into the implementation method because it's hard to know the return type from the API.

Instead, return CompletableFuture for this method.

Reply:
@codelipenghui Fixed.